crypto/ppccap.c

```diff
@@ -106,16 +111,17 @@ void poly1305_blocks_fpu(void *ctx, const unsigned char *inp, size_t len,
                          unsigned int padbit);
 void poly1305_emit_fpu(void *ctx, unsigned char mac[16],
                        const unsigned int nonce[4]);
+int poly1305_init(void *ctx, const unsigned char key[16], void *func[2]);
 int poly1305_init(void *ctx, const unsigned char key[16], void *func[2])
 {
     if (sizeof(size_t) == 4 && (OPENSSL_ppccap_P & PPC_FPU)) {
         poly1305_init_fpu(ctx, key);
-        func[0] = poly1305_blocks_fpu;
-        func[1] = poly1305_emit_fpu;
+        func[0] = (void*)(uintptr_t)poly1305_blocks_fpu;
+        func[1] = (void*)(uintptr_t)poly1305_emit_fpu;
```

Comment:
I'd be more comfortable with `void (*func[2])()` than with cast...

Reply:
Well, when this function is called the protoype is
`int poly1305_init(void *ctx, const unsigned char key[16], void *func)`
and what is passed is a pointer to `struct {
        poly1305_blocks_f blocks;
        poly1305_emit_f emit;
    };`
this is certainly an aliasing issue.
I see no justification for aliasing a void (*func[2])() with an anonymous struct.

Reply:
I'd say another PR.

Reply:
In all other cases the structure is filled by assembly subroutine. But assembler has no notion of structures, so ABI "steps in" which says that it *can* be viewed as two-element array. Just in case for reference, why doesn't one initialize it in assembly even on PPC. This is because references to global variables and how they handled in assembly differ too much among platforms, it's just too much.